### DISCUSSION TITLE
How to use setDimensions function to reduce the size of all objects inside at the same time?

### DISCUSSION THREAD

**Discussion by kwdiwt** [fabricjs/fabric.js#8353] [Q&A] [answered]
http://jsfiddle.net/17h942ob/7/
```
const canvas = new fabric.Canvas('c');
const rect = new fabric.Rect({
   left: 100,
   top: 100,
   fill: 'red',
   width: 30,
   height: 30
 });
canvas.add(rect);
canvas.renderAll();

const el = document.getElementById('toggle');
el.addEventListener("click", function(e) {
  canvas.setDimensions({
    width: 150,
    height: 150
  });
}, false);
```
I want to have the same effect as setting the background size in css.
or how to scale overall.

**Comment by iocoker** [comment, marked as answer]
Setting dimension on the canvas will resize the canvas. You may want to `setZoom`?? like this: [http://jsfiddle.net/uzahjrm2/](http://jsfiddle.net/uzahjrm2/)

Otherwise, you will have to get each objects and `setDimension` on each and also offset their positions `(top, left)` appropriately.

**Reply by ShaMan123** [reply]
changing the `viewportTransform` is the correct approach as @iocoker suggests

**Reply by kwdiwt** [reply]
> Setting dimension on the canvas will resize the canvas. You may want to `setZoom`?? like this: http://jsfiddle.net/uzahjrm2/
> 
> Otherwise, you will have to get each objects and `setDimension` on each and also offset their positions `(top, left)` appropriately.

oh! yes, It's this setZoom.   thx
